### PR TITLE
gtk4: update to 4.1.2

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
I'm mostly interested in the re-new activate signal in `GtkCheckButton`

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

File: Gsk-4.0.gir [hunk 1]
```diff
@@ -573,6 +574,23 @@ that the gradient points up. Color stops are then added clockwise.</doc>
           </parameter>
         </parameters>
       </constructor>
+      <method name="get_angle" c:identifier="gsk_conic_gradient_node_get_angle">
```
Comment (sdroege) [inline Gsk-4.0.gir:577]: Misses a `Since` marker in the C docs

File: Gsk-4.0.gir [hunk 2]
```diff
@@ -1830,11 +1860,37 @@ points and color stops, and render that into the area given by @bounds.</doc>
...
+        </parameter>
+      </parameters>
+    </function-macro>
+    <function-macro name="NGL_RENDERER_GET_CLASS" c:identifier="GSK_NGL_RENDERER_GET_CLASS" introspectable="0">
+      <parameters>
+        <parameter name="obj">
+        </parameter>
+      </parameters>
+    </function-macro>
+    <class name="NglRenderer" c:symbol-prefix="ngl_renderer" c:type="GskNglRenderer" parent="Renderer" glib:type-name="GskNglRenderer" glib:get-type="gsk_ngl_renderer_get_type" glib:type-struct="NglRendererClass">
```
Comment (sdroege) [inline Gsk-4.0.gir:1886]: Since marker missing too for the whole type

**Comment by sdroege** [comment]
Same basically for all new API here. That's sad and should be fixed in C really :)

**Comment by bilelmoussaoui** [comment]
Fixed those upstream https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3268. let's wait till the next release to pull the gir files.

**Comment by sdroege** [comment]
Hard to review with all the docs and `<attribute>` noise there... do you have a PR that shows how the result after `gir` looks like?

**Comment by bilelmoussaoui** [comment]
Updated to 4.1.2 which contains fixes for the missing version annotations, it also has  a ton of docs fixes (which is the reason of the very large diff)

**Comment by bilelmoussaoui** [comment]
> Hard to review with all the docs and `<attribute>` noise there... do you have a PR that shows how the result after `gir` looks like?

https://github.com/gtk-rs/gtk4-rs/pull/217

**Comment by sdroege** [comment]
Seems fine to me after the discussions in the other PR!

**Pull request merged**